File: vueapp/src/models/account.js

function Account (accountId, accountName)
{
    this.AccountId          = accountId || 0
    this.AccountName        = accountName || ''
    this.AccountEmail       = ''      
    this.AccountPhone       = '' 
    this.StreetAddress      = '' 
    this.City               = '' 
    this.StateProvince      = '' 
    this.PostalCode         = ''
    this.IsInvoiceAccount     = false
    this.IsAutoPay            = false
    this.IsActive           = false 
}

export
{
    Account
}

File: coreLogic/Models/Account.cs
using coreLogic.Interfaces;
using coreLogic.Models.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace coreApi.Models;

public class Account : IAuditable
{
    [Required]
	public int AccountId { get; set; }

	[Required, MaxLength(50)]
    public string AccountName { get; set; } = "";

	[Required, EmailAddress]
    public string AccountEmail { get; set; } = "";

	[Required, Phone]
	public string AccountPhone { get; set; } = "";

	[Required, MaxLength(50)]
	public string StreetAddress { get; set; } = "";

	[Required, MaxLength(50)]
	public string City { get; set; } = "";

	[Required, MaxLength(2)]
	public string StateProvince { get; set; } = "";

	[Required, MaxLength(10)]
	public string PostalCode { get; set; } = "";

	public bool IsInvoice { get; set; }

	public bool IsAutoPay { get; set; }

	public bool IsActive { get; set; } = true;

	public DateTime DateCreated { get; set; }
	
	public DateTime DateModified { get; set; }
	
	public int CreatorId { get; set; }
	
	public int ModifierId { get; set; }

	[NotMapped]
	public string CreatorName { get; set; }

	[NotMapped]
	public string ModifierName { get; set; }

	public override string ToString() => $"{AccountName} AccountId: {AccountId}";
}